### ISSUE TITLE
JOSE pre-submission enquiry : Matlab to Octave Porting Guide

### ISSUE THREAD

**Issue by raamana** [openjournals/jose-old#29]
**Project repository URL:** To be uploaded

Dear Editors, 

JOSE is a great idea, and I support its mission. Congratulations!

We have developed a comprehensive step-by-step guide to encourage academics to share and open source their Matlab code by making it easier to port it to Octave. As the open science and open source movement is picking up steam, the desire to convert existing Matlab codebases (which still occupies a large fraction of biomedical research) into more open Octave is significant. We believe our manual based on covering a large and complex neuroscience pipeline, along with setup and validation scripts , would make future conversions much easier and tolerable than would be otherwise. Does this sound suitable for publication in JOSE?

The guide would be in Markdown and published as a website cross-referring to a code repository.

Thanks,
Pradeep

**Comment by moorepants** [comment]
Pradeep,

Your guide sounds interesting, but as stated I don't see how it is related to education. See http://jose.theoj.org/about to learn about the papers we accept. For example, your submission would have to be either: 
- open educational software tools
- open-source learning modules
If you can describe how your guide meets the criteria we'd be happy to reconsider.

**Comment by raamana** [comment]
Thanks for the response. This is an enquiry to find out its suitability. It is fine if this was judged to be out of scope, but I'd appreciate considering it. 

JOSE website has TLDR saying "your course or lesson content must contain or use code to teach". The  guide would be targeted towards undergrads and graduate students (or anyone else) entering an academic lab relying on Matlab codebases and **teaching how to make Matlab tools more open source** (convert to Octave) and verifying the scientific accuracy of the conversion. As it stands, this is a daunting effort for many newcomers, often resulting in subtle errors they are not always equipped to identify or debug.

This can be seen as similar to http://jose.theoj.org/papers/51d19b9c42e3115a548acd4a1cad377b (teaching how to create HTML presentations), with a lot more focus on science, numerical computing and informatics. And as I noted before, we are at the verge of lot of Matlab code getting converted to Octave to join the open science movement in academia, and this guide would likely be well received.

**Comment by moorepants** [comment]
The EMACs Reveal software is an "open educational software tool". Is your submission a "open educational software tool" or an "open-source learning module"? Please describe how it is one or the other of these. It sounds like the latter to me. If so, then does it fall under this description: "Examples include Jupyter notebooks or plaintext/markup language documents like LaTeX, R Markdown, and ReST for course/lesson content and associated notes, with embedded or associated code snippets/programs."

**Comment by raamana** [comment]
You are right - it is indeed open-source learning module, and it does fall under the description you noted: markup docs with embedded and associate snippets and code.

**Comment by moorepants** [comment]
I suggest you submit for a more detailed review then.

**Comment by raamana** [comment]
Just to clarify, are you approving of this, and suggesting I make a formal submission? I'd love to do that, but only after I get an approval from of the editors here that it is suitable, and will be reviewed.

**Comment by moorepants** [comment]
I am an editor and it seems to be that your submission may fit based on our criteria. Without going through pre-review with an actual submission I can say no more. I suggest submitting your work to the journal. There are no guarantees of making it through pre-review or the follow up peer review.

**Comment by raamana** [comment]
Great, thanks. I understand the submission must be accepted by reviewers.

One more question, are we allowed to share the submission on my webiste/blog during peer review? I read something on JOSE website to the effect of "it must not be available elsewhere", so want to clarify. Everything will be public in the future anyways, and my blog post will eventually refer to the JOSE paper and ask them cite it if they used it.

**Comment by moorepants** [comment]
> One more question, are we allowed to share the submission on my webiste/blog during peer review?

This should be fine. I think it refers to that it should not be formally published by another journal.
